src/main/java/org/aksw/simba/squirrel/frontier/impl/FrontierImpl.java

```diff
@@ -84,14 +102,30 @@ public FrontierImpl(KnownUriFilter knownUriFilter, UriQueue queue) {
      *            {@link UriQueue} used to manage the URIs that should be
      *            crawled.
      */
-    public FrontierImpl(KnownUriFilter knownUriFilter, UriQueue queue, GraphLogger graphLogger) {
+    public FrontierImpl(KnownUriFilter knownUriFilter, UriQueue queue, GraphLogger graphLogger, boolean doesRecrawling) {
         this.knownUriFilter = knownUriFilter;
         this.queue = queue;
         this.uriProcessor = new UriProcessor();
         this.graphLogger = graphLogger;
 
         this.queue.open();
         this.knownUriFilter.open();
+        this.doesRecrawling = doesRecrawling;
+
+        if (doesRecrawling) {
+            timerRecrawling = new Timer();
+            timerRecrawling.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    List<CrawleableUri> urisToRecrawl = knownUriFilter.getOutdatedUris();
+                    urisToRecrawl.forEach(uri -> queue.addUri(uriProcessor.recognizeUriType(uri)));
+                }
+            }, 0, TIMER_PERIOD);
+        }
+    }
+
+    public UriQueue getQueue() {
+        return queue;
     }
```

Comment:
It might be better to "extend" the visibility of the `queue` method to `public` instead of coming up with a second method :thinking: 

This would also resolve the problem of the missing javadoc :wink:

Reply:
@MichaelRoeder I cannot find that second method :)